infra/Pos/Communication/BiP.hs

```diff
@@ -11,62 +11,23 @@ module Pos.Communication.BiP
 import           Universum
 
 import qualified Data.ByteString            as BS
-import qualified Data.ByteString.Lazy       as LBS
-import qualified Data.Store                 as Store
-import qualified Network.Transport.Internal as NT (decodeWord32, encodeWord32)
 
 import           Node.Message.Class         (Serializable (..))
 import           Node.Message.Decoder       (Decoder (..))
 
-import           Pos.Binary.Class           (Bi (..), encode, label)
+import           Pos.Binary.Class           (Bi (..), serialize, decodeFull)
 
 data BiP = BiP
 
 instance  Bi t => Serializable BiP t where
```

Comment:
Ok, this is not correct. We need to transform a CBOR incremental `Decoder` into our own incremental `Decoder`. This will then get run by the network `recv` action (which will do the overall length checking).